transport/src/main/java/io/netty/channel/nio/NioEventLoop.java

```diff
@@ -448,16 +512,17 @@ protected Runnable pollTask() {
         return task;
     }
 
-    private void processSelectedKeysPlain(Set<SelectionKey> selectedKeys) {
+    private int processSelectedKeysPlain(Set<SelectionKey> selectedKeys) {
         // check if the set is empty and if so just return to not create garbage by
         // creating a new Iterator every time even if there is nothing to process.
         // See https://github.com/netty/netty/issues/597
         if (selectedKeys.isEmpty()) {
-            return;
+            return 0;
         }
 
         Iterator<SelectionKey> i = selectedKeys.iterator();
-        for (;;) {
+        int processed = 0;
+        for (;; processed ++) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) This loop's stop condition tests "" but the incrementer updates "processed". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1994)